api-2.2/src/main/java/org/openmrs/htmlformentryui/ConditionElement.java

```diff
@@ -0,0 +1,309 @@
+		if (classCache != null) {
+			requiredClasses.add(classCache);
+		}
+		
+		for (ConceptClass cc : requiredClasses) {
+			initialConcepts.addAll(Context.getConceptService().getConceptsByClass(cc));
+		}
+		
+		conditionName = new ConceptSearchAutocompleteWidget(new ArrayList<Concept>(initialConcepts), requiredClasses);
+		String conditionNameTextInputId = context.registerWidget(conditionName);
+		conditionNameErrorWidget = new ErrorWidget();
+		context.registerErrorWidget(conditionName, conditionNameErrorWidget);
+		
+		StringBuilder ret = new StringBuilder();
+		ret.append(conditionName.generateHtml(context));
+		if (context.getMode() != Mode.VIEW) {
+			ret.append(conditionNameErrorWidget.generateHtml(context));
+		}
+		ret.append("\n<script>jq('#" + conditionNameTextInputId + "').attr('placeholder',");
+		ret.append(" '" + mss.getMessage("coreapps.conditionui.condition") + "');\n");
+		// Add support for non-coded concept values.
+		// This a hack to let the autocomplete widget accept values that aren't part of the concept list.
+		ret.append("jq('#" + conditionNameTextInputId + "').blur(function(e){\n");
+		ret.append("     var valueAttr = jq('#" + conditionNameTextInputId + "_hid" + "').attr('value');\n");
+		ret.append("     if(valueAttr === \"ERROR\"){\n");
+		ret.append("        jq('#" + conditionNameTextInputId + "_hid" + "').attr('value', '');\n");
+		ret.append("     }\n");
+		ret.append("});\n");				
+		ret.append("</script>\n");
+		return ret.toString();
+	}
+	
+	private String createConditionStatusWidgets(FormEntryContext context) {
+		if (mss == null) {
+			mss = Context.getMessageSourceService();		
+		}
+		Option active = new Option(mss.getMessage("coreapps.conditionui.active.label"), "active", false);
+		Option inactive = new Option(mss.getMessage("coreapps.conditionui.inactive.label"), "inactive", false);
+		conditionStatusWidget = new RadioButtonsWidget();
+		conditionStatusErrorWidget = new ErrorWidget();
+		
+		conditionStatusWidget.addOption(active);
+		conditionStatusWidget.addOption(inactive);
+		conditionStatusWidget.setInitialValue("active");
+		conditionStatusWidget.setAnswerSeparator("");
+		String radioGroupName = context.registerWidget(conditionStatusWidget);
+		context.registerErrorWidget(conditionStatusWidget, conditionStatusErrorWidget);
+
+		StringBuilder sb = new StringBuilder();
+		sb.append(conditionStatusErrorWidget.generateHtml(context));
+		
+		sb.append("<script>");		
+		sb.append("jq(\"input[name='" + radioGroupName + "']\").change(function(e){\n" +
+				"    if($(this).val() == 'active') {\n" + 
+				"		jq('#endDatePicker').hide(); \n" +
+				"    } else {\n" + 
+				"		 jq('#endDatePicker').show(); \n" +
+				"    }\n" + 
+				"\n" + 
+				"});");
+		sb.append("</script>");		
+		return conditionStatusWidget.generateHtml(context) + sb.toString();	
+	}
+	
+	private String createConditionDateWidgets(FormEntryContext context) {
```

Comment:
I would rather see a `protected` method named `htmlForConditionDateWidgets(FormEntryContext)`.

Reply:
Why don't we have it `private` ?